### PR TITLE
Add `dune build src/lib` to unit test runner

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR is a follow-on from #12098, fixing the remaining compilation errors when `dune build src/lib` is run in the repo.

This adds the command to the 'dev unit test' CI job, which will ensure that this compilation doesn't break again over time.

Mostly, this involves dune tricks to avoid building targets that we don't need in certain situations (e.g. not attempting to build a dynamic `.so` library from the rust code when compiling javascript tests).

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

**Review by mimoo — APPROVED** [review]

**Review by lk86 — APPROVED** [review]

File: src/lib/consensus/dune [hunk 1]
```diff
@@ -106,4 +106,5 @@
  (libraries core_kernel signature_lib mina_state mina_block consensus prover blockchain_snark logger.file_system)
  (preprocessor_deps "../../config.mlh")
  (instrumentation (backend bisect_ppx))
- (preprocess (pps ppx_version ppx_jane)))
+ (preprocess (pps ppx_version ppx_jane))
+ (enabled_if false))
```
Comment (psteckler) [inline src/lib/consensus/dune:110]: worth a comment to say why this is disabled, and when it might be re-enabled?

**Comment by psteckler** [comment]
Maybe put comments in the empty files indicating why they're there.

**Review by psteckler — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**